### DISCUSSION TITLE
Protected commands

### DISCUSSION THREAD

**Discussion by madprops** [fish-shell/fish-shell#10811] [Ideas]
Some commands are heavy, as in they take a long time, or they start doing expensive operations.

Might be useful if we could register specific commands to require a confirm by quickly pressing enter again.

For instance says I can register `./utils/bundle.rb && ./utils/check.sh` to the protected commands list.

Then when I execute that command (manually) I need to press Enter again in the next 500ms or the command gets cancelled.

So I am required to double press Enter to confirm.

**Comment by zanchey** [comment]
This can be implemented as a custom binding - see https://github.com/fish-shell/fish-shell/pull/9761
